File: frontend/src/types/feature.ts
export type FeatureStatus = "backlog" | "planned" | "in_progress" | "done" | "cut"
export type MvpClassification = "mvp" | "v1" | "v2" | "nice_to_have"
export type PriorityLevel = "critical" | "high" | "medium" | "low"

export type Feature = {
  id: string
  projectId: string
  specId: string | null
  title: string
  description: string
  priorityScore: number
  effortEstimate: number | null
  status: FeatureStatus
  parentFeatureId: string | null
  sortOrder: number
  mvpClassification: MvpClassification
  dependencies: string[]
  childFeatures: Feature[]
  createdAt: string
  updatedAt: string
}

export type FeatureCreate = {
  title: string
  description?: string
  parentFeatureId?: string
  mvpClassification?: MvpClassification
}

export type FeatureUpdate = Partial<FeatureCreate> & {
  status?: FeatureStatus
  sortOrder?: number
  priorityScore?: number
}

File: backend/app/schemas/feature.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class FeatureCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=300)
    description: str | None = None
    spec_id: str | None = None
    parent_feature_id: str | None = None
    priority_score: float | None = None
    effort_estimate: str | None = None
    mvp_classification: str | None = None
    sort_order: int = 0


class FeatureUpdate(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=300)
    description: str | None = None
    priority_score: float | None = None
    effort_estimate: str | None = None
    status: str | None = None
    mvp_classification: str | None = None
    sort_order: int | None = None


class FeatureResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    spec_id: str | None
    title: str
    description: str | None
    priority_score: float | None
    effort_estimate: str | None
    status: str
    parent_feature_id: str | None
    sort_order: int
    mvp_classification: str | None
    created_at: datetime
    updated_at: datetime
